test/client-errors.js

```diff
@@ -519,7 +517,9 @@ test('socket fail while writing request body', (t) => {
     body.push('asd')
 
     client.on('connect', () => {
-      client.socket.destroy('kaboom')
+      process.nextTick(() => {
+        client.socket.destroy('kaboom')
+      })
```

Comment:
We resume the queue *after* emitting connect, thus we need to defer the destroy to keep the test valid.